### PR TITLE
feat(hogql): select from s3

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

We need support `select from function()` for two cases:

- selecting from `s3(user uploaded large file)` (data warehouse is working on this CC @timgl)
- selecting from the numbers table (blocking moving trends to a hogql insight)

## Changes

Implements the first of the two, and leaves some building blocks for the second.

I did not expose the `s3` method directly, instead created a `S3Table` class that can be used as a hogql database table.

I also had to change to `readonly=2` from `=1` to [allow temporary tables](https://clickhouse.com/docs/en/operations/settings/permissions-for-queries). That flag also gives access to `SET something=1` queries, but HogQL can only create "SELECT" queries anyway, so we're good. If someone finds a way to inject arbitrary text into the generated SQL, then we likely have bigger issues than this.

## How did you test this code?

Wrote some tests. Added a hidden `aapl_stock` table that you can select from to try it out. It's from the ClickHouse docs.

### CODE REVIEW

File: posthog/hogql/database/s3_table.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+from typing import Dict
+
+from posthog.hogql.database.models import TableFunction, DatabaseField
+from posthog.hogql.escape_sql import escape_hogql_identifier
+
+
+class S3Table(TableFunction):
```
Comment (timgl) [inline posthog/hogql/database/s3_table.py:7]: I called this `ExternalTable`, as we'll support GCP/Azure and potentially others. Don't feel very strongly about this.

File: docker/clickhouse/config.xml [hunk 2]
```diff
@@ -677,6 +677,7 @@
             Regexps are not aligned: don't forget to add ^ and $. Also don't forget to escape dot (.) metacharacter
             (forgetting to do so is a common source of error).
         -->
+        <host>datasets-documentation.s3.eu-west-3.amazonaws.com</host>
```
Comment (timgl) [inline docker/clickhouse/config.xml:680]:
```suggestion
            <host_regexp>^s3.*\.amazonaws\.com$</host_regexp>
            <host_regexp>^.*\.googleapis\.com$</host_regexp>
```

Makes it consistent with prod

**Review by timgl — APPROVED** [review]

**Comment by mariusandra** [comment]
I did plenty of changes since the last review, and this should be good to go now, assuming CI agrees. Tagging @Twixes and @thmsobrmlr for even more 👀.

@timgl btw the `s3.*\.amazonaws\.com` enabled host didn't match the `https://datasets-documentation.s3.eu-west-3.amazonaws.com/aapl_stock.csv` URL I was using. I moved the bucket from the domain to the path for my URL and didn't change anything here, but perhaps `.*\.s3.*\.amazonaws\.com` needs to be added to the list as well?

**Comment by timgl** [comment]
Ah yep I'll add that to my PR and update prod

**Review by Twixes — COMMENTED** [review]
Looks great to me

**Pull request merged**